### ISSUE TITLE
Doxyfile environment/quote mishandling in list values

### ISSUE THREAD

**Issue by therealchjones** [doxygen/doxygen#9362]
**Describe the bug**
When environment variables whose values contain spaces are used in configuration tags that expect list values, quotes are stripped inappropriately and the resulting values are not valid.

**Expected behavior**
Values containing spaces can be included by quoting them, just as literal values can be and environment variables in configuration tags that expect a string can be

**To Reproduce**

Doxyfile:
```
# Doxyfile 1.9.4

OUTPUT_DIRECTORY = "$(output_dir)"
INPUT = "$(input_dir)"
EXCLUDE = "Directory With Spaces/input/exclude" "$(input_dir)/exclude2"
```

output:
```
% output_dir="Directory With Spaces/output" input_dir="Directory With Spaces/input" doxygen -x
# Difference with default Doxyfile 1.9.4
OUTPUT_DIRECTORY       = "Directory With Spaces/output"
INPUT                  = Directory \
                         With \
                         Spaces/input
EXCLUDE                = "Directory With Spaces/input/exclude" \
                         Directory \
                         With \
                         Spaces/input/exclude2
```

**Version**
1.9.4, via Homebrew, on zsh on macOS 12.4

**Comment by therealchjones** [comment]
For anyone referencing this issue, the silly workaround I'm using is:
```
% input_dir="$(echo "" | sed 's/\//\\\//g')"
sed "s/\$(SRCDIR)/$SRCDIR/g" "$CONFIGDIR"/Doxyfile \
	| DOXYGEN_OUTPUT="$(setdir "$BUILDDIR/doxygen")" doxygen -x -

**Comment by albert-github** [comment]
It is all a matter of quoting and in the different shells the quotes can give different problem.
I think a better workaround would be to use either:
- `input_dir='"Directory With Spaces/input"'`

or
- `input_dir="\"Directory With Spaces/input\""`

problem is that e.g. `INPUT` can have multiple entries as input and these are separated by white space, so it is for doxygen not possible to see whether with e.g. `"Directory With Spaces/input"` the user means the directories:
- `Directory`
- `With`
- `Spaces/input`

or one directory:
- `Directory With Spaces/input`

as the shell already removed the outer `"`.

I think the problem is usage.

**Comment by therealchjones** [comment]
As you can see from the example, doxygen has no problem interpreting quoted values in a list, only choking when they’re interpolated from environment variables. Similarly, it has no problem quoting environment variables properly, only choking when they’re in a list. And unfortunately the shell-quoting workaround doesn’t work, as doxygen splits the list value at quote that’s included in the environment variable rather than including any appended strings. Try your workarounds with my example Doxyfile and you’ll see they don’t work. 

I believe the issue is that the Doxygen config file parser is handling quotes and environment variable interpolation in the wrong order for list values, but I’m not certain of that.

**Comment by albert-github** [comment]
Looking back at e.g. `INPUT = "$(input_dir)"` I think that here doxygen has indeed a small problem that needs further investigation in respect to the moment when the quotes are stripped versus the moment of the evaluation of the environment variable.

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #9366

**Comment by albert-github** [comment]
Code has been integrated in master on GitHub (please don't close the issue as this will be done at the moment of an official release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.5.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
